src/blockchain.js

```diff
@@ -267,6 +267,11 @@ class Blockchain {
     // signatures are correct
     for (let i = 1; i < this.chain.length; i++) {
       const currentBlock = this.chain[i];
+      const previousBlock = this.chain[i - 1];
+
+      if (previousBlock.hash !== currentBlock.previousHash) {
```

Comment:
Could you use the `calculateHash()` method here instead?
Otherwise it would still be possible to manually set a correct hash, even if it doesn't match with the contents of the block.

Reply:
Right

Reply:
Hello @Savjee ,
I think that is already validated at line 280.

For the previous block the hash value was validated in the line 280 in the previous iteration. For the current block, it will validated at line 280 in the current iteration. The first block is the génesis one, I don’t think there is a problem with that.

Adding calculateHash() does not affect too much the time complexity of this validation method, but better not call it if not necessary right?

Reply:
Maybe one thing we are missing in this method is validate that hash match the configured difficulty, so that if the attacker changes the block information and recalculate all the hashes, they are still  compliant with the difficulty restriction.

If I am correct we may creat a different task/pr for that.

Reply:
You're right, we don't need to recalculate it in the loop. Nice catch!

Yes, the difficulty check is a good idea. And should indeed happen in a separate PR. I'll merge this one in first ;)